### ISSUE TITLE
Make statement type list better accessible in FactoidBuilder

### ISSUE THREAD

**Issue by GVogeler** [acdh-oeaw/apis-instance-manmax#20]
Gendering Maximilian: "Teile des Menüs sind leider nicht sichtbar - also auf der rechten Seite (z.B. “Anderes”)"

**Comment by GVogeler** [comment]
Gendering Maximilian: "Bei der Hierarchisierung von Statements: Das Menü ist jetzt schon sehr aufbläht und deshalb verwirrend (z.B. familiäre Verbindung ist ganz oben, später kommt noch Elternschaft, Geschwisterverhältnis etc., ich denke, damit die Daten wirklich sinnvoll strukturiert eingegeben werden, sollte das klarer geordnet sein.)"

**Comment by richardhadden** [comment]
There are some overflow problems in standard APIS menu — I can't really find a good way to fix this. But people should be using FactoidBuilder.

As for "logical structure"... I'd be interested to know if there was one! (Originally, I experimented with having everything in one menu, with the hierarchical structure represented, but it takes up so much space on the screen; so I went with this current "grouping"... which is subjective, I admit.)

**Comment by GVogeler** [comment]
On the logical structure:
1. sort order of the menus is currently not understandable by the user. Sorting them alphabetically by German display name? 
2. Introducing deeper hierarchy or sorting them by some additional logical flag could increase overview
3. having something autocomplete style (including the comments?) could improve handling of larger lists if types

**Comment by richardhadden** [comment]
1. Alphabetical order is a good idea (at least it will look a little nicer)
2. The hierarchy exists, but statement types are grouped by "Subject Group", so it's not really possible to show more than one hierarchy (as groups cut across hierarchy) — other option would be to have the hierarchy first, then group by "Subject Group" (maybe with some icon for each subject?). Tree diagrams type things waste a lot of screen space though
3. This is a nice idea. I can probably implement this without much work, in fact

**Comment by richardhadden** [comment]
https://github.com/acdh-oeaw/apis-instance-manmax/assets/77681667/2ebd167a-94cb-414a-9817-087ab3cfd725

Implements 1 and 3!

(Searching also looks at the help text as well as the name)

**Comment by richardhadden** [comment]
https://github.com/acdh-oeaw/apis-instance-manmax/assets/77681667/1070d78b-20a6-44cf-8a3c-9b5977fce956

Implements 1 and 3!

(Searching also looks at the help text as well as the name)

**Comment by richardhadden** [comment]
https://github.com/acdh-oeaw/apis-instance-manmax/assets/77681667/fb8a3666-2300-4f53-968b-79b68505436b

A couple more behaviours:
1. Right clicking on any Statement type shows the description
2. If there's only one Statement type found by filtering, it's automatically selected on just pressing Enter

**Comment by GVogeler** [comment]
As for tree style visualisations: if they are collapsed they don't need much space, do they?

**Comment by richardhadden** [comment]
True... I'll try to implement this, though I suspect arbitrarily hiding things wouldn't help usability.

Another possible idea is to allow users (or sub-projects?) to customise which statement-types they see. Most simply, this could just be done by hard-coding lists of "allowed types" for each subproject, and a selector menu somewhere. If we want users to be able to change this themselves, then we'd have to think about database models and interface to manage it (and then these lists would have to be dynamically loaded, instead of just being some interface configuration)

**Comment by GVogeler** [comment]
> True... I'll try to implement this, though I suspect arbitrarily hiding things wouldn't help usability.

Before implementing this, I think it would be worth to deploy the two changes.
